tuf/utils/utils.go

```diff
@@ -150,3 +154,87 @@ func ConsistentName(role string, hashSha256 []byte) string {
 	}
 	return role
 }
+
+// ErrWrongLength indicates the length was different to that expected
+var ErrWrongLength = errors.New("wrong length")
+
+// ErrWrongHash indicates the hash was different to that expected
+type ErrWrongHash struct {
+	Type     string
+	Expected []byte
+	Actual   []byte
+}
+
+// Error implements error interface
+func (e ErrWrongHash) Error() string {
+	return fmt.Sprintf("wrong %s hash, expected %#x got %#x", e.Type, e.Expected, e.Actual)
+}
+
+// ErrNoCommonHash indicates the metadata did not provide any hashes this
+// client recognizes
+type ErrNoCommonHash struct {
+	Expected data.Hashes
+	Actual   data.Hashes
+}
+
+// Error implements error interface
+func (e ErrNoCommonHash) Error() string {
+	types := func(a data.Hashes) []string {
+		t := make([]string, 0, len(a))
+		for typ := range a {
+			t = append(t, typ)
+		}
+		return t
+	}
+	return fmt.Sprintf("no common hash function, expected one of %s, got %s", types(e.Expected), types(e.Actual))
+}
+
+// ErrUnknownHashAlgorithm - client was ashed to use a hash algorithm
+// it is not familiar with
+type ErrUnknownHashAlgorithm struct {
+	Name string
+}
+
+// Error implements error interface
+func (e ErrUnknownHashAlgorithm) Error() string {
+	return fmt.Sprintf("unknown hash algorithm: %s", e.Name)
+}
+
+// PassphraseFunc type for func that request a passphrase
+type PassphraseFunc func(role string, confirm bool) ([]byte, error)
+
+// FileMetaEqual checks whether 2 FileMeta objects are consistent with eachother
+func FileMetaEqual(actual data.FileMeta, expected data.FileMeta) error {
```

Comment:
non-blocking: given this seems to be unused and with all the new and improved hash checking utils in `data.CheckHashes` and `data.CompareMultiHashes`, should we just delete this function?